web/client/epics/geostories.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const Rx = require('rxjs');
+const { MAPS_LIST_LOADING, ATTRIBUTE_UPDATED} = require('../actions/maps');
+const { MAP_DELETED, MAP_METADATA_UPDATED } = require('../actions/maps');
+const { SAVED: GEOSTORY_SAVED } = require('../actions/geostory');
+
+const { SEARCH_GEOSTORIES, DELETE_GEOSTORY, GEOSTORY_DELETED, RELOAD, searchGeostories, geostoriesListLoaded, geostoryDeleted, geostoriesLoading } = require('../actions/geostories');
+const { searchParamsSelector, searchTextSelector, totalCountSelector} = require('../selectors/geostories');
+const GeoStoreApi = require('../api/GeoStoreDAO');
+const { wrapStartStop } = require('../observables/epics');
+const {error} = require('../actions/notifications');
+
+const {deleteResource} = require('../api/persistence');
+
+const calculateNewParams = state => {
+    const totalCount = totalCountSelector(state);
+    const {start, limit, ...params} = searchParamsSelector(state) || {};
+    if (start === totalCount - 1) {
+        return {
+            start: Math.max(0, start - limit),
+            limit
+        };
+    }
+    return {
+        start, limit, ...params
+    };
+};
+
+module.exports = {
+    searchGeostoriesOnMapSearch: action$ =>
+        action$.ofType(MAPS_LIST_LOADING)
```

Comment:
Please check this MAPS_LIST_LOADING

Reply:
Dashboard and geostories firs search is thrown by MAPS_LIST_LOADING